interface/ApfpBlas.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <gmp.h>
+
+#include <functional>
+
+#include "ApfpInterfaceType.h"
+
+namespace apfp {
+
+enum BlasError : int {
```

Comment:
Why not make this an `enum class` also?

Reply:
Unfortunately BLAS has the named errors (positive integers) and also argument errors (negative integers). 
Ex passing too small of `LDA` to `syrk` returns `-5`. This is a standard BLAS convention so I think we should keep it

Reply:
Doesn't this still work with an enum class, exactly as you did it here?